components/job-orchestration/job_orchestration/retention/utils.py

```
@@ -0,0 +1,151 @@
+    This class support recovering from a previous failure by reading previously persisted targets
+    from a recovery file. The user is expected to explicitly call `persists_new_targets` to persist
+    any new targets on to the disk for fault-tolerance purposes.
+
+    :param recovery_file_path: Path to the file used for recovering and persisting targets.
+    :raises ValueError: If the recovery path exists but is not a file.
+    """
+
+    def __init__(self, recovery_file_path: pathlib.Path):
+        self._recovery_file_path: pathlib.Path = recovery_file_path
+        self._targets: Set[str] = set()
+        self._targets_to_persist: List[str] = list()
```

Pylint (3.3.7)

[refactor] Consider using [] instead of list()

(R1734)